src/plugins/data/common/exports/export_csv.tsx

```diff
@@ -22,34 +23,54 @@ interface CSVOptions {
   escapeFormulaValues: boolean;
   formatFactory: FormatFactory;
   raw?: boolean;
-  columnsSorting?: string[];
+  sortedColumns?: string[];
+  columnSorting?: EuiDataGridColumnSortingConfig[];
```

Comment:
Here we pass both the sorted columns and now any column sorting config the user has defined.

Reply:
the `EuiDataGridColumnSortingConfig` is an array of `id` and `direction`, can we just use this instead of having both the sortedColumns and this array?
We can sort that configs array the same way as the `sortedColums`

Reply:
Are saying to combine them such that the order of the `columnSorting` matches that of the `sortedColumns`?

```ts
// do this...
const columnSorting = [{ id: 'col-a', direction: 'none'}, { id: 'col-b', direction: 'desc'}];
// instead of this...
const sortedColumns = ['col-a', 'col-b'];
const columnSorting = [{ id: 'col-b', direction: 'desc'}];
```

In that case I don't think so. The `columnSorting` is already an array who's order indicates the hierarchy of multiple filters. This is the same behavior in eui. Even though the current Table vis only supports a singular column sort, I don't think it would be a good idea.

Even in terms of usability of this function, I think it is much easier on the user as is rather than having to combine them, let alone understand what the order means.